core/spec/models/spree/order_credit_sorter_spec.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Spree::OrderCreditSorter, type: :model do
+  let(:order) { create(:order) }
+  let(:credits) { Spree::StoreCredit.all }
+  let(:sorter) { described_class.new(credits, order) }
+
+  describe '#sort' do
+    subject { sorter.sort }
+
+    let!(:primary_credit) { create(:store_credit) }
+    let!(:secondary_credit) { create(:store_credit, credit_type: create(:secondary_credit_type)) }
+    
+    it 'returns the expected credits' do
+      expect(subject).to contain_exactly(primary_credit, secondary_credit)
+    end
+
+    it 'uses the default sorting order' do
+      expect(credits).to receive(:order_by_priority)
```

Comment:
Maybe a matter of taste, but I'd prefer to see the actual behavior being tested instead of the implementation details. I'd assert on the returned order instead.

Reply:
I can do that - will update!